public/vislib/marker_types/base_marker.js

```diff
@@ -342,11 +348,44 @@ define(function (require) {
         } else {
           this._legendColors = reds5;
         }
-
         this._legendQuantizer = d3.scale.quantize().domain(quantizeDomain).range(this._legendColors);
+
+      } else { // Dynamic - Uneven
+        // A legend scale that will create uneven ranges for the legend in an attempt
+        // to split the map features uniformly across the ranges.  Useful when data is unevenly
+        // distributed across the minimum - maximum range.
+        let features = this.geoJson.features;
+        let featureLength = features.length;
+        features.sort(function(x, y) {
+          return d3.ascending(x.properties.value, y.properties.value);
+        });
+
+        let bottomCutoff = 1;
```

Comment:
Lets use the same logic to select the color scale for linear and uneven dynamic scales.